python/foglamp/plugins/south/cc2650poll/cc2650poll.py

```diff
@@ -0,0 +1,259 @@
+         'type': 'string',
+         'default': 'cc2650poll'
+    },
+    'pollInterval': {
+        'description': 'The interval between poll calls to the device poll routine expressed in milliseconds.',
+        'type': 'integer',
+        'default': '500'
+    },
+    'bluetoothAddress': {
+        'description': 'Bluetooth MAC address',
+        'type': 'string',
+        'default': 'B0:91:22:EA:79:04'
+    }
+}
+
+_LOGGER = logger.setup(__name__, level=20)
+
+sensortag_characteristics = characteristics
+
+
+def plugin_info():
+    """ Returns information about the plugin.
+
+    Args:
+    Returns:
+        dict: plugin information
+    Raises:
+    """
+
+    return {
+        'name': 'Poll plugin',
+        'version': '1.0',
+        'mode': 'poll',
+        'type': 'device',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG
+    }
+
+
+def plugin_init(config):
+    """ Initialise the plugin.
+
+    Args:
+        config: JSON configuration document for the device configuration category
+    Returns:
+        handle: JSON object to be used in future calls to the plugin
+    Raises:
+    """
+    global sensortag_characteristics
+
+    bluetooth_adr = config['bluetoothAddress']['value']
+    tag = SensorTagCC2650(bluetooth_adr)
+
+    # The GATT table can change for different firmware revisions, so it is important to do a proper characteristic
+    # discovery rather than hard-coding the attribute handles.
+    for char in sensortag_characteristics.keys():
+        for type in ['data', 'configuration', 'period']:
+            handle = tag.get_char_handle(sensortag_characteristics[char][type]['uuid'])
+            sensortag_characteristics[char][type]['handle'] = handle
+
+    # print(json.dumps(sensortag_characteristics))
+
+    data = copy.deepcopy(config)
+    data['characteristics'] = sensortag_characteristics
+    data['bluetooth_adr'] = bluetooth_adr
+
+    _LOGGER.info('SensorTagCC2650 {} Polling initialized'.format(bluetooth_adr))
+
+    return data
+
+
+def plugin_poll(handle):
+    """ Extracts data from the sensor and returns it in a JSON document as a Python dict.
+
+    Available for poll mode only.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+        returns a sensor reading in a JSON document, as a Python dict, if it is available
+        None - If no reading is available
+    Raises:
+        DataRetrievalError
+    """
+
+    time_stamp = str(datetime.datetime.now(tz=datetime.timezone.utc))
+    data = {
+        'asset': 'TI sensortag',
+        'timestamp': time_stamp,
+        'key': str(uuid.uuid4()),
+        'readings': {}
+    }
+
+    try:
+        bluetooth_adr = handle['bluetooth_adr']
+        object_temp_celsius = None
+        ambient_temp_celsius = None
+        lux_luminance = None
+        rel_humidity = None
+        rel_temperature = None
+        bar_pressure = None
+        movement = None
```

Comment:
L#125 to L#129 - Local variables are not used. More info with PEP8 please inspect